src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests/Resources/SingleString_ClassNameCSharpAsync_NS1.NS2/Resources.Designer.cs

```diff
@@ -0,0 +1,24 @@
+﻿// <auto-generated/>
+
+#nullable enable
+using System.Reflection;
+
+namespace TestProject
+{
+    internal static class Resources { }
+}
+namespace NS1
+{
+    internal static partial class NS2
+    {
+        private static global::System.Resources.ResourceManager? s_resourceManager;
+        public static global::System.Resources.ResourceManager ResourceManager => s_resourceManager ?? (s_resourceManager = new global::System.Resources.ResourceManager(typeof(TestProject.Resources)));
```

Comment:
❓ Is this correct? Just want to make sure.

Reply:
There's a comment that explains why this empty type is created:

https://github.com/dotnet/roslyn-analyzers/blob/7aba2caa536e168f93da3a23e427887521f57a4d/src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator/AbstractResxGenerator.cs#L573-L575

I'm not sure why we can't use the other type though.